packages/frontend/src/pages/scaling-projects/props/getProjectHeader.ts

```diff
@@ -0,0 +1,55 @@
+import { Layer2 } from '@l2beat/config'
+import { ActivityApiResponse, TvlApiResponse } from '@l2beat/types'
+
+import { formatLargeNumber } from '../../../utils'
+import { getTpsDaily } from '../../../utils/activity/getTpsDaily'
+import { getTpsWeeklyChange } from '../../../utils/activity/getTpsWeeklyChange'
+import { getTransactionMonthlyCount } from '../../../utils/activity/getTransactionWeeklyCount'
+import { getTvlWithChange } from '../../../utils/tvl/getTvlWitchChange'
+import { formatUSD } from '../../../utils/utils'
+import { ProjectHeaderProps } from '../view/ProjectHeader'
+
+export function getProjectHeader(
+  project: Layer2,
+  tvlApiResponse: TvlApiResponse,
+  activityApiResponse?: ActivityApiResponse,
+): ProjectHeaderProps {
+  const { tvl, tvlWeeklyChange } = getTvlWithChange(tvlApiResponse, project)
+
+  const activityData =
+    activityApiResponse?.projects[project.id.toString()]?.data
+  const tpsDaily = getTpsDaily(activityData)
+  const tpsWeeklyChange = getTpsWeeklyChange(activityData)
+  const transactionMonthlyCount = getTransactionMonthlyCount(activityData)
+
+  return {
+    icon: `/icons/${project.display.slug}.png`,
+    title: project.display.name,
+    titleLength: getTitleLength(project.display.name),
+    tvl: formatUSD(tvl),
+    tvlWeeklyChange,
+    tpsDaily: tpsDaily?.toFixed(2) ?? '',
+    tpsWeeklyChange,
+    transactionMonthlyCount:
+      transactionMonthlyCount !== undefined
+        ? formatLargeNumber(transactionMonthlyCount)
+        : undefined,
+    purpose: project.display.purpose,
+    technology: project.technology.category,
+  }
+}
+
+function getTitleLength(name: string): 'long' | 'very-long' | undefined {
+  switch (name) {
+    case 'Optimism':
+    case 'rhino.fi':
+    case 'Immutable X':
+      return 'long'
+    case 'OMG Network':
+    case 'Layer2.Finance':
+    case 'ZKSwap V2':
+    case 'Polygon Hermez':
+    case 'Metis Andromeda':
+      return 'very-long'
+  }
+}
```

Comment:
It is duplicated with the file getProjectHeader which should be deleted after we turn on the activity feature flag. This will not be used anywhere else, so IMO it makes sense to keep this function inside this file. WYT?

Reply:
Ok :)